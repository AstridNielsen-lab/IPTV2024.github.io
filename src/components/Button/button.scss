@import '../../static/sass/abstracts/functions';

.n-button {
  top: 40rem;
  cursor: pointer;
  font-size: 1.6rem;
  outline: none;
  border: none;
  font-weight: 700;
  border-radius: 5px;
  padding-left: 3.5rem;
  padding-right: 3.5rem;
  margin-right: 1rem;
  padding-top: 1rem;
  padding-bottom: 1rem;

  @include responsive(phone) {
    top: 24rem;
  }

  &.with-icon {
    fill: #fff;
    margin-right: 1rem;
    display: flex;

    & > * {
      width: 15px;
      height: 15px;
      fill: #fff;
      margin-right: 1rem;
    }
  }

  &.Primary {
    color: #000;
    background-color: #fff;
    transition: all 0.2s;
    box-shadow: 0 1px 2px rgba(0, 0, 0, 0.3);

    & > * {
      fill: #000;
    }
  }

  &.Secondary {
    background-color: rgba(109, 109, 110, 0.7);
    color: #fff;
  }
}
