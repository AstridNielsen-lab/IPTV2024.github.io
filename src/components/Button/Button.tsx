import './button.scss'

export enum ButtonType {
  Primary,
  Secondary,
}

interface IButton {
  label?: string
  onClick?: () => void
  buttonType: ButtonType
  Icon?: JSX.Element
}

const Button = ({ label, onClick, buttonType, Icon }: IButton) => {
  const getIconClassName = (): '' | ' with-icon' => {
    return Icon ? ' with-icon' : ''
  }

  const getLabelClassName = (): '' | ' with-label' => {
    return label ? ' with-label' : ''
  }

  const getPrimaryClassName = (): string => {
    return ButtonType[buttonType ? buttonType : ButtonType.Primary]
  }

  return (
    <button
      className={`n-button${getIconClassName()}${getLabelClassName()} ${getPrimaryClassName()}`}
      onClick={onClick}
    >
      {Icon && Icon}
      {label}
    </button>
  )
}

export default Button
