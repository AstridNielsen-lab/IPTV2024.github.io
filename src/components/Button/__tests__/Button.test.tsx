import { Primary, Secondary } from './Button.stories'
import { render } from '@testing-library/react'

describe('Button', (): void => {
  test('should render primary button', (): void => {
    const { container } = render(<Primary {...Primary.args} />)
    expect(container).toMatchSnapshot()
  })

  test('should render secondary button', (): void => {
    const { container } = render(<Secondary {...Secondary.args} />)
    expect(container).toMatchSnapshot()
  })

  test('should render primary button with icon', (): void => {
    const { container } = render(<Secondary {...Secondary.args} />)
    expect(container).toMatchSnapshot()
  })
})
