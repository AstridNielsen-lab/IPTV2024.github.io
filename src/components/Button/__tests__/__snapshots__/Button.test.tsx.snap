// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render primary button 1`] = `
<div>
  <button
    class="n-button Primary"
  >
    Primary
  </button>
</div>
`;

exports[`Button should render secondary button 1`] = `
<div>
  <button
    class="n-button Secondary"
  >
    Secondary
  </button>
</div>
`;
